### ISSUE TITLE
DataTable: Column resize is not working properly.

### ISSUE THREAD

**Issue by dmacannuco** [primefaces/primereact#4471]
### Describe the bug

The resize of the columns of the DataTable component is not working correctly when we have updated to the latest version of the 9.5.0 primereact library.

The error is the same as the example on the official documentation page (https://primereact.org/datatable/)

The error is reproduced in both columnResizeMode.

- In "expand" mode the column place is not correct and also the resize of one column affects the rest.
[error_expand_mode.webm](https://github.com/primefaces/primereact/assets/36630670/3b3494f3-5795-46bb-a751-61f7cb922402)

- In "fit" mode, the resize directly does not work.
[error_fit_mode.webm](https://github.com/primefaces/primereact/assets/36630670/3324f777-d9a6-4d7d-bf3a-ee0032b03d98)

The most curious thing is that if we try to run the code in the sandbox, in both cases, it seems to work correctly in a simplified project.

However, the same thing happens in our project as is happening in the examples on the documentation website.

### PrimeReact version

9.5.0

### React version

18.x

### Language

ES6

### Build / Runtime

Create React App (CRA)

**Comment by melloware** [comment]
Agreed that is weird as its working in StackBlitz: https://stackblitz.com/edit/react-11cbe3?file=src%2FApp.js

**Comment by bu3alwa** [comment]
Yes so this is the same thing happening in my app. So in dev mode in nextjs this does not have an issue and works properly but in production mode it seems something happens maybe during the optimization process that breaks the behavior of the resize. 

I haven't been able to debug this further.

**Comment by bu3alwa** [comment]
@dmacannuco what was the previous version you were at before you upgraded?

**Comment by melloware** [comment]
@bu3alwa this probably has to do with REact 18 Strict mode.  Using the PR Showcase just change `next.config.js` to..

```js
reactStrictMode: false, //process.env.NODE_ENV === 'production' ? false : true,
```

And I bet you see the same behavior now as production mode.    That will help narrow down the issue to improper hook usage.

**Comment by bu3alwa** [comment]
This issue might be a bit on my lower priority for next week but Ill check the week after and let you know what I can find.

**Comment by dmacannuco** [comment]
> @dmacannuco what was the previous version you were at before you upgraded?

The version from which we have updated has been from 8.3.0.
In our case, we have the same problem in both development and production mode.

**Comment by bu3alwa** [comment]
ok I know whats causing it but I don't know where its happening and why. So on the `div` wrapper for a datatable on html attribute shows up `pr_id_1` as a unique identifier for a datatable. The css that gets added when resizing also references that same id. But on first load that is missing. If you trigger a refresh with HMR you'll see it show up. 

I am unsure where in the code that is set so if you can let me know I can debug further.

**Comment by shuklabhisekh** [comment]
Hey @bu3alwa, I've already fixed it and created a PR(https://github.com/primefaces/primereact/pull/4410) for this issue. It has been approved by @melloware but waiting to get approved by other reviewers and get merged. You can check it and test it you want.

**Comment by bu3alwa** [comment]
I think yours would run into the same issue. On load the attribute is missing. 

https://github.com/primefaces/primereact/blob/eac31f5cfee793150f81bda885a80dff3432cf9f/components/lib/datatable/DataTable.js#L1343

This is where the id is generated but I have to check how its passed to the prop. Ill check more tomorrow

**Comment by bu3alwa** [comment]
ok, this was bothering me so 

```
    useUpdateEffect(() => {
        elementRef.current.setAttribute(attributeSelector.current, '');
    }
    , [attributeSelector.current])
```

This would be a fix but I dunno if it has any sideeffects. 

@melloware do you think this is safe to add? or would that break anything else?

**Comment by melloware** [comment]
Let me look at this. I thought you couldn't have a Hook listen on a ref.current?   Refs don't trigger re-renders right?

**Comment by melloware** [comment]
OK fixed submitted for this issue. It had to do with mounting the component properly.

**Comment by melloware** [comment]
Fixed in 9.6.2 if you want to give it a try

**Comment by dmacannuco** [comment]
I just tested it and everything seems to work fine.
Thanks for everything.
Regards.
